internal/controller/cluster/service.go

```diff
@@ -162,6 +162,30 @@ func (r *SingleClusterReconciler) createOrUpdateSTSLoadBalancerSvc() error {
 	return r.updateLBService(service, &servicePort, loadBalancer)
 }
 
+func (r *SingleClusterReconciler) deleteLBServiceIfPresent(svcName, svcNamespace string) error {
+	service := &corev1.Service{}
+	svcNamespacedName := types.NamespacedName{Name: svcName, Namespace: svcNamespace}
+
+	if err := r.Client.Get(context.TODO(), svcNamespacedName, service); err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("LoadBalancer is not configured. Skipping...")
+
+			return nil
+		}
+
+		return fmt.Errorf("failed to get loadbalancer service %s: %v", svcName, err)
+	}
+
+	if err := r.Client.Delete(context.TODO(), service); err != nil {
+		return fmt.Errorf("failed to delete loadbalancer service %s: %v", svcName, err)
+	}
```

Comment:
Instead of making 2 API calls, direct call Delete and check for IsNotFound error. Adjust the logging accordingly.
See if you do the same for `deletePodService` as well.